### ISSUE TITLE
If I click 'Download Files' from the Save menu, UmpleOnline then hangs

### ISSUE THREAD

**Issue by TimLethbridge** [umple/umple#1554]
After successfully downloading files, UmpleOnline hangs!

On the main server the download happens, then the freeze occurs. On my machine I get 

```
Fatal error: Uncaught Error: Class 'ZipArchive' not found in /Users/tcl/unix/umple/umpleonline/scripts/tab_control.php:92 Stack trace: #0 {main} thrown in /Users/tcl/unix/umple/umpleonline/scripts/tab_control.php on line 92
```

**Comment by WeiqiMaster** [comment]
I've encountered this error on my local server too. My solution is to install the php-zip package.

**Comment by TimLethbridge** [comment]
That doesn't explain the crash on the server though. Please check on your machine to download all files from the SAVE menu and then continue using UmpleOnline. Does it work? Please test several times. Make sure you have some Umple code in the text window.

**Comment by WeiqiMaster** [comment]
I just tested all examples on https://cruise.eecs.uottawa.ca/umpleonline/ and no problem found.

**Comment by TimLethbridge** [comment]
The problem still persists on the main server and is severe. However, I was surprised by your response, so I have dug deeper. It is browser-specific. The problem occurs in
a) Firefox
b) Safari

It does not seem to appear in
c) Opera
d) Google Chrome

Here is what you do in the affected browsers.

1. Display an example (or write any code yourself)
2. in the SAVE & Load menu click 'Download files'
3. At this point you should see the 'Generate Java' pink icon with a 'crossout' in it, and also if you try to generate code nothing happens, or if you edit code it won't be reflected in the diagram

**Comment by TimLethbridge** [comment]
Progress on this in pr #1558

**Comment by TimLethbridge** [comment]
Fixed by #1558
